pkg/config/lima_config_applier.go

```diff
@@ -188,3 +199,50 @@ func hasUserModeEmulationInstallationScript(limaCfg *limayaml.LimaYAML) (int, bo
 
 	return scriptIdx, hasCrossArchToolInstallationScript
 }
+
+func toggleSoci(limaCfg *limayaml.LimaYAML, enabled bool, sociVersion string, arch string) {
+	idx, hasScript := hasSociInstallationScript(limaCfg)
+	fname := fmt.Sprintf("soci-snapshotter-%s-linux-%s.tar.gz", sociVersion, arch)
+	sociDownloadUrl := fmt.Sprintf("https://github.com/awslabs/soci-snapshotter/releases/download/v%s/%s", sociVersion, fname)
+	if !hasScript && enabled {
+		limaCfg.Provision = append(limaCfg.Provision, limayaml.Provision{
+			Mode: "system",
+			Script: fmt.Sprintf(`%s
+if [ ! -f /usr/local/bin/soci ]; then
+    #download soci
+	export config=etc/containerd/config.toml
+    curl -OL "%s" >> $config
+    #move to usr/local/bin
+    tar -C /usr/local/bin -xvf %s ./soci ./soci-snapshotter-grpc
```

Comment:
`./` doesn't work. Need to remove it. Follow what you had in nerdctl. https://github.com/containerd/nerdctl/blob/1a7e89dcc9edbee903adbb49756f2575b8450c53/Dockerfile#L290